cli/azd/pkg/infra/provisioning/provider.go

```diff
@@ -52,10 +54,9 @@ type StateResult struct {
 
 type Provider interface {
 	Name() string
-	Initialize(ctx context.Context, projectPath string, options Options) error
+	Initialize(ctx context.Context, projectPath string, options Options, rg osutil.ExpandableString) error
```

Comment:
I initially considered trying to have `Initialize` take the entire `project.ProjectConfig` object so it could grab this property, but that doesn't work because it creates an import cycle between the `project` and `provisioning` packages.

Sort of a bummer that the `resourceGroup` property doesn't hang off of the `provisioning.Options` type.  It probably would have made more sense to nest that under `infra` vs it being a top level property.